WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductRowViewModel.swift

```diff
@@ -63,7 +67,18 @@ final class ProductRowViewModel: ObservableObject, Identifiable, Equatable {
 
     /// Quantity of product in the order
     ///
-    @Published var quantity: Int64 = 1
+    @Published private(set) var quantity: Decimal = 1 {
+        willSet {
+            onQuantityChange?(newValue)
+        }
```

Comment:
Since these view models already expose a quantity publisher, is there a reason not to subscribe to that publisher and send a completion block instead? 

Normally I think we should use completion blocks when trying to communicate with the outside world,  AKA UIKit, but I think that we should rely on `@Binding` and `@Published` variables as much as possible when dealing with view models internally!

Reply:
I used a completion block because I couldn't find a way to subscribe to the publisher while still connecting the quantity change to the right element in the `productRows` array in the new order. I might be missing something about how to set this up, so I'll take another look. (Anyway I need to solve the problem where the quantity changes get mixed up when the same product is added more than once! Thanks for noticing that, too.)